Lib/test/test__interpreters.py

```diff
@@ -0,0 +1,506 @@
+import contextlib
+import os
+import os.path
+import shutil
+import tempfile
+from textwrap import dedent, indent
+import threading
+import unittest
+
+from test import support
+from test.support import script_helper
+
+interpreters = support.import_module('_interpreters')
+
+
+SCRIPT_THREADED_INTERP = """\
+from textwrap import dedent
+import threading
+import _interpreters
+def f():
+    _interpreters.run_string(id, dedent('''
```

Comment:
I don't understand where the `id` variable is coming from... is it the built-in `id` function?